app/graphql/concerns/fancy_mutation.rb

```diff
@@ -0,0 +1,196 @@
+# Provides a bunch of helpers for our standard mutation structure, making consistency easy!
+#
+# @example
+#   class Mutations::Post::Like < Mutations::Base
+#     # Pass a description here to be used in the GraphQL schema
+#     description 'Like a post'
+#     # Build your input type here (set arguments)
+#     input do
+#       argument :postId, ID,
+#         required: true,
+#         description: 'The post to like'
+#     end
+#     # Your result type should generally expose what changed so clients can update their cache
+#     result Types::Post
+#     # Build the union of errors which you may return
+#     errors Types::Errors::NotAuthorized,
+#       Types::Errors::NotAuthenticated,
+#       Types::Errors::Conflict
+#
+#     def resolve(post_id:)
+#       return errors << Types::Errors::NotAuthenticated.build unless context[:current_user]
+#       # In most cases you'll want to delegate to a Mutator class so the logic can be unit-tested.
+#       PostMutator.like(post: Post.find(post_id), user: context[:current_user])
+#     rescue ActiveRecord::RecordInvalid => e
+#       errors << Types::Errors::Conflict.build
+#     rescue Pundit::NotAuthorizedError => e
+#       errors << Types::Errors::NotAuthorized.build
+#     end
+#   end
+module FancyMutation
+  extend ActiveSupport::Concern
+
+  class WarningsPresent < StandardError; end
+
+  module PrependedMethods
+    # Wraps the resolve method with some support. When ignore_warnings is false or not set, it will
+    # rollback when there are warnings. In all cases, the return value of the resolve method will be
+    # wrapped as the :result field, so that the mutation always returns the correct structure. When
+    # the return value is the errors list, we ignore it, so that we won't accidentally try to treat
+    # it as the result type.
+    # @param ignore_warnings [Boolean] Whether to ignore warnings or not
+    def resolve(ignore_warnings: false, **args)
```

Comment:
I switched this to a prepended `#resolve` instead of `#resolve_with_support` so that we're not touching any private interfaces. Additionally, this should ensure that code in `#ready?` and `#authorized?` will run outside this transaction, which is important for rate-limiting (or would be, if we used Postgres as the backing store for that)